lib/zclient.c

```diff
@@ -1622,6 +1622,48 @@ static bool zapi_decode_prefix(struct stream *s, struct prefix *p)
 	return false;
 }
 
+static void zapi_encode_sockunion(struct stream *s, const union sockunion *su)
+{
+	int family = sockunion_family(su);
+	size_t addrlen = family2addrsize(family);
+
+	stream_putc(s, (uint8_t)family);
+
+	/*
+	 * degenerate case if unknown length
+	 */
+	if (!addrlen)
```

Comment:
Shouldn't this check be before we put something to the stream?

Reply:
Changed to assert - if we're trying to encode an unknown family, it's a bug.

[Hmm, I meant to submit this comment yesterday but must not have clicked the button...]